### PR TITLE
Origin/vt pair rate 190819

### PR DESCRIPTION
**Pull request by tien7668**

### CODE REVIEW

File: src/js/reducers/globalReducer.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import constants from '../services/constants';
+import Cookies from 'js-cookie';
```
Comment (andrewsource147) [inline src/js/reducers/globalReducer.js]: Why we need that lib, when we already has this func in common?

File: src/js/containers/LimitOrder/LimitOrder.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ import * as limitOrderActions from "../../actions/limitOrderActions"
 import constants from "../../services/constants"
 import {LimitOrderBody} from "../LimitOrder"
 import limitOrderServices from "../../services/limit_order";
-import { isUserLogin } from "../../utils/common";
+import {default as common, isUserLogin} from "../../utils/common";
```
Comment (andrewsource147) [inline src/js/containers/LimitOrder/LimitOrder.js]: This is weird?

File: src/js/containers/LimitOrder/LimitOrder.js [hunk 3]
```diff
@@ -128,7 +128,15 @@ export default class LimitOrder extends React.Component {
 
     return {sourceTokenSymbol, sourceToken, destTokenSymbol, destToken}
   }
-
+  async fetchFavoritePairsIfLoggedIn(){
+    if (isUserLogin) {
+      limitOrderServices.getFavoritePairs().then(
```
Comment (andrewsource147) [inline src/js/containers/LimitOrder/LimitOrder.js]: If its async func, please use await with this, code will be clearer

File: src/js/containers/LimitOrder/LimitOrderModals/OrderDetailsModal.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
+import React, { Component } from 'react';
+import { connect } from "react-redux";
+import { getTranslate } from "react-localize-redux";
+import ReactTable from "react-table";
+
+import { Modal } from "../../../components/CommonElement";
+import { LimitOrderSubmit } from "../../LimitOrder";
+import {getFormattedDate} from "../../../utils/common";
+import * as converters from "../../../utils/converter";
+import {displayNumberWithDot} from "../../../utils/converter";
```
Comment (andrewsource147) [inline src/js/containers/LimitOrder/LimitOrderModals/OrderDetailsModal.js]: This import is redundant, please use import * as converters from "../../../utils/converter";

File: src/js/containers/LimitOrder/LimitOrderTable.js [hunk 5]
```diff
@@ -623,17 +694,31 @@ export default class LimitOrderTable extends Component {
       )
     } else if (title === "status") {
       return (
-        <Dropdown active={this.state.statusFilterVisible} onHide={e => this.togglingStatusFilter()}>
-          <div>
-            <span>{(this.props.translate("limit_order.status") || "Status").toUpperCase()}</span>
-            <div className="drop-down">
-              <img src={require("../../../assets/img/v3/price_drop_down.svg")}/>
+          <Dropdown active={this.state.statusFilterVisible} onHide={e => this.togglingStatusFilter()}>
```
Comment (andrewsource147) [inline src/js/containers/LimitOrder/LimitOrderTable.js]: When we use () => func() and when we use e => func(). I see many places are different

File: src/js/containers/LimitOrder/LimitOrderTable.js [hunk 6]
```diff
@@ -827,5 +934,6 @@ export default class LimitOrderTable extends Component {
 LimitOrderTable.propTypes = {
```
Comment (andrewsource147) [inline src/js/containers/LimitOrder/LimitOrderTable.js]: I dont see this style in our project. If its possible, please update as other files

File: src/js/containers/LimitOrder/QuoteMarket/QuoteMarket.js [hunk 7]
```diff
@@ -88,7 +86,7 @@ export default class QuoteMarket extends React.Component{
   search(quotes){
     const { current_search, current_sort_index, current_sort_dsc, pairs } = this.state
     const { currentQuote } = this.props
-
+    console.log("zx", currentQuote, quotes)
```
Comment (andrewsource147) [inline src/js/containers/LimitOrder/QuoteMarket/QuoteMarket.js]: Please remove its comment

File: src/js/containers/LimitOrder/withFavorite.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { connect } from "react-redux"
+import * as common from "../../utils/common";
+import limitOrderServices from "../../services/limit_order";
+import * as limitOrderActions from "../../actions/limitOrderActions";
```
Comment (andrewsource147) [inline src/js/containers/LimitOrder/withFavorite.js]: Can you sort import as action -> services -> utils

File: src/js/containers/LimitOrder/LimitOrderTable.js (hunk 5 above)
Comment (tien7668) [reply src/js/containers/LimitOrder/LimitOrderTable.js]: With only 1 param, the parentheses can be removed

File: src/js/containers/LimitOrder/LimitOrderTable.js (hunk 6 above)
Comment (tien7668) [reply src/js/containers/LimitOrder/LimitOrderTable.js]: I see. It's from legacy code

File: src/js/reducers/globalReducer.js (hunk 1 above)
Comment (tien7668) [reply src/js/reducers/globalReducer.js]: Updated, using common instead.

File: src/js/containers/LimitOrder/LimitOrder.js (hunk 2 above)
Comment (tien7668) [reply src/js/containers/LimitOrder/LimitOrder.js]: Updated

File: src/js/containers/LimitOrder/LimitOrder.js (hunk 3 above)
Comment (tien7668) [reply src/js/containers/LimitOrder/LimitOrder.js]: Updated, using await

File: src/js/containers/LimitOrder/LimitOrderModals/OrderDetailsModal.js (hunk 4 above)
Comment (tien7668) [reply src/js/containers/LimitOrder/LimitOrderModals/OrderDetailsModal.js]: Updated, using converters only

File: src/js/containers/LimitOrder/QuoteMarket/QuoteMarket.js (hunk 7 above)
Comment (tien7668) [reply src/js/containers/LimitOrder/QuoteMarket/QuoteMarket.js]: Updated

File: src/js/containers/LimitOrder/withFavorite.js (hunk 8 above)
Comment (tien7668) [reply src/js/containers/LimitOrder/withFavorite.js]: Updated

File: package.json [hunk 9]
```diff
@@ -29,6 +29,7 @@
     "ethereumjs-wallet": "^0.6.0",
     "hdkey": "^1.1.0",
     "history": "^4.7.2",
+    "js-cookie": "^2.2.1",
```
Comment (andrewsource147) [inline package.json]: If we don't use it, please remove

File: src/js/containers/LimitOrder/LimitOrderModals/OrderDetailsModal.js [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+import { getTranslate } from "react-localize-redux";
+import ReactTable from "react-table";
+
+import { Modal } from "../../../components/CommonElement";
+import { LimitOrderSubmit } from "../../LimitOrder";
+import {getFormattedDate} from "../../../utils/common";
+import * as converters from "../../../utils/converter";
+import {formatNumber} from "../../../utils/converter";
+import {multiplyOfTwoNumber} from "../../../utils/converter";
+import {subOfTwoNumber} from "../../../utils/converter";
```
Comment (andrewsource147) [inline src/js/containers/LimitOrder/LimitOrderModals/OrderDetailsModal.js]: Please removing formatNumber, multiplyOfTwoNumber, subOfTwoNumber

File: package.json (hunk 9 above)
Comment (tien7668) [reply package.json]: Removed

File: src/js/containers/LimitOrder/LimitOrderModals/OrderDetailsModal.js (hunk 10 above)
Comment (tien7668) [reply src/js/containers/LimitOrder/LimitOrderModals/OrderDetailsModal.js]: Removed

**Pull request merged**